test/new-e2e/tests/windows/base_agent_installer_suite.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023-present Datadog, Inc.
+
+// Package windows contains the code to run the e2e tests on Windows
+package windows
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	platformCommon "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common"
+	windowsAgent "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common/agent"
+	infraCommon "github.com/DataDog/test-infra-definitions/common"
+	"path/filepath"
+	"strings"
+)
+
+// BaseAgentInstallerSuite is a base class for the Windows Agent installer suites
+type BaseAgentInstallerSuite[Env any] struct {
+	e2e.BaseSuite[Env]
+
+	AgentPackage *windowsAgent.Package
+	OutputDir    string
+}
+
+// InstallAgent installs the Agent on a given Windows host. It will pass all the parameters to the MSI installer.
+func (b *BaseAgentInstallerSuite[Env]) InstallAgent(host *components.RemoteHost, options ...windowsAgent.InstallAgentOption) (string, error) {
+	b.T().Helper()
+	p, err := infraCommon.ApplyOption(&windowsAgent.InstallAgentParams{
+		InstallLogFile: "install.log",
+	}, options)
+	if err != nil {
+		return "", err
+	}
+
+	if p.Package == nil {
+		return "", fmt.Errorf("missing agent package to install")
+	}
+
+	var args []string
+
+	if p.AgentUser != "" {
+		args = append(args, fmt.Sprintf("DDAGENTUSER_NAME=%s", p.AgentUser))
+	}
+	if p.AgentUserPassword != "" {
+		args = append(args, fmt.Sprintf("DDAGENTUSER_PASSWORD=%s", p.AgentUserPassword))
+	}
+	if p.Site != "" {
+		args = append(args, fmt.Sprintf("SITE=%s", p.Site))
+	}
+	if p.APIKey != "" {
+		args = append(args, fmt.Sprintf("APIKEY=%s", p.APIKey))
+	}
+	if p.DdURL != "" {
+		args = append(args, fmt.Sprintf("DD_URL=%s", p.DdURL))
+	}
+
+	remoteMSIPath, err := common.GetTemporaryFile(host)
+	if err != nil {
+		return "", err
+	}
+	err = common.PutOrDownloadFile(host, p.Package.URL, remoteMSIPath)
+	if err != nil {
+		return "", err
+	}
+
+	return remoteMSIPath, common.InstallMSI(host, remoteMSIPath, strings.Join(args, " "), filepath.Join(b.OutputDir, p.InstallLogFile))
+}
+
+// NewTestClientForHost creates a new TestClient for a given host.
+func (b *BaseAgentInstallerSuite[Env]) NewTestClientForHost(host *components.RemoteHost) *platformCommon.TestClient {
+	// We could bring the code from NewWindowsTestClient here
+	return platformCommon.NewWindowsTestClient(b.T(), host)
+}
+
+// SetupSuite overrides the base SetupSuite to perform some additional setups like setting the package to install
+// or getting the output directory for the install logs.
+func (b *BaseAgentInstallerSuite[Env]) SetupSuite() {
+	b.BaseSuite.SetupSuite()
+
+	var err error
+	b.OutputDir, err = runner.GetTestOutputDir(runner.GetProfile(), b.T())
```

Comment:
This should be moved to `SetupTest` so each test gets its own output dir.

Using a suite field for the output dir might be problematic for tabular tests/subtests, if they just refer to `OutputDir` then they will use the same directory may overwrite each others output files.

It is convenient for top level tests though so I think it's worth keeping as long as we keep that in mind. Maybe with a comment on the `OutputDir` struct field?